python/cudf/cudf/tests/test_categorical.py

```diff
@@ -677,3 +679,28 @@ def test_add_categories_mixed_error():
 
     with pytest.raises(TypeError):
         gds.cat.add_categories(["a", "bd", "ef"])
+
+
+def test_categorical_equal_compare():
```

Comment:
Could this test be moved to `test_dtypes`? That's where the other tests are for `CategoricalDtype`, including the tests for equality suggested below

Reply:
Yup - moving.